src/main/java/ch/njol/skript/conditions/CondIsLeftHanded.java

```diff
@@ -0,0 +1,86 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Mob;
+import org.bukkit.inventory.MainHand;
+
+@Name("Left Handed")
+@Description({
+	"Checks if living entities or players are left or right handed. Armor stands are neither right nor left-handed.",
+	"Paper 1.17.1+ is required for non-player entities."
+})
+@Examples({
+	"on damage of player:",
+		"\tif victim is left handed:",
+			"\t\tcancel event"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper 1.17.1+ (entities)")
+public class CondIsLeftHanded extends PropertyCondition<LivingEntity> {
+
+	private static final boolean CAN_USE_ENTITIES = Skript.methodExists(Mob.class, "isLeftHanded");
+
+	static {
+		if (CAN_USE_ENTITIES) {
+			register(CondIsLeftHanded.class, PropertyType.BE, "(:left|right)[ |-]handed", "livingentities");
+		} else {
+			register(CondIsLeftHanded.class, PropertyType.BE, "(:left|right)[ |-]handed", "players");
+		}
+	}
+
+	private boolean leftHand;
+
+	@Override
+	public boolean init(Expression[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		leftHand = parseResult.hasTag("left");
+		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	}
+
+	@Override
+	public boolean check(LivingEntity livingEntity) {
+		// check if entity is a mob and if the method exists
+		if (CAN_USE_ENTITIES && livingEntity instanceof Mob)
+			return ((Mob) livingEntity).isLeftHanded() == leftHand;
+
+		// check if entity is a player
+		if (livingEntity instanceof HumanEntity)
+			return ((HumanEntity) livingEntity).getMainHand() == (leftHand ? MainHand.LEFT : MainHand.RIGHT);
```

Comment:
You don't need to use a ternary here: you can compare `hand = left XOR !expected`

So for example `mainHand == LEFT ^ !leftHand`.

| mainHand == LEFT | leftHand | !leftHand | mainHand == LEFT ^ !leftHand | Expected |
|------------------|----------|-----------|------------------------------|----------|
| T                | T        | F         | T                            | T        |
| T                | F        | T         | F                            | F        |
| F                | T        | F         | F                            | F        |
| F                | F        | T         | T                            | T        |

Reply:
That having been said, it might even be preferable to store the hand enum you're looking for rather than a boolean for lefthanded.

Reply:
oh storing the enum would be much nicer, good point